src/space-between/internal.tsx

```diff
@@ -8,12 +8,16 @@ import flattenChildren from 'react-keyed-flatten-children';
 import { SpaceBetweenProps } from './interfaces';
 import { InternalBaseComponentProps } from '../internal/hooks/use-base-component';
 
-type InternalSpaceBetweenProps = SpaceBetweenProps & InternalBaseComponentProps;
+type InternalSpaceBetweenProps = SpaceBetweenProps &
+  InternalBaseComponentProps & {
+    asList?: boolean;
```

Comment:
The challenge is that space-between supports two different directions and also responsiveness. We want to keep that no matter if list is used or not. Using Box won't be enough, we can use tokens but that would require replicating complex space-between CSS everywhere. I rather like making it a public API but it is not necessary to fast-forward it. It would be great to use it internally at first to better understand what list props we'd like to support.